docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,518 @@ aliases:
+  if err != nil {
+    return err
+  }
+
+  // If the returned events list has no items, no event was created
+  if len(events.Items) != 1 {
+    fmt.Printf(
+      "No event was created. "+
+        "Does project %s subscribe to events of source %s and type %s?\n",
+      projectID,
+      event.Source,
+      event.Type,
+    )
+    return nil
+  }
+
+  // The Brigade event was successfully created!
+  fmt.Printf(
+    "Event created with ID %s for project %s\n",
+    events.Items[0].ID,
+    projectID,
+  )
+  return nil
+}
 ```
-$ docker push technosophos/example-cron
-```
 
-Now we create our third (and last) file: a CronJob definition. Our `cron.yaml`
-should look something like this:
+### Subscribing a project to events from the example gateway
+
+In order to utilize events from the example gateway, we'll need a Brigade
+project that subscribes to the corresponding event source
+(`brigade.sh/example-gateway`) and event type (`create-event`). We'll also
+define an event handler that handles these events and utilizes the attached
+payload.
+
+Here's the project definition file.  Note the `spec.eventSubscriptions` section
+and the default `brigade.ts` script which contains our event handler:
 
 ```yaml
-apiVersion: batch/v1beta1
-kind: CronJob
+apiVersion: brigade.sh/v2-beta
+kind: Project
 metadata:
-  name: example-cron-gateway
-  labels:
-    heritage: brigade
-    component: gateway
+  id: example-gateway-project
+description: |-
+  An example project that subscribes to events from an example gateway
 spec:
-  schedule: "*/1 * * * *"
-  jobTemplate:
-    spec:
-      template:
-        spec:
-          restartPolicy: OnFailure
-          containers:
-            - name: cron-example
-              image: technosophos/example-cron:latest
-              imagePullPolicy: IfNotPresent
+  eventSubscriptions:
+  - source: brigade.sh/example-gateway
+    types:
+      - create-event
+  workerTemplate:
+    logLevel: DEBUG
+    defaultConfigFiles:
+      brigade.ts: |
+        import { events } from "@brigadecore/brigadier"
+
+        events.on("brigade.sh/example-gateway", "create-event", async event => {
+          console.log("Hello, " + event.payload + "!")
+        })
+
+        events.process()
+```
+
+We can save this to `project.yaml` and create it in Brigade via the following
+command:
+
+```console
+$ brig project create --file project.yaml
+```
+
+### Running the gateway
+
+Now that we have a project subscribing to events from this gateway, we're ready
+to build and run the example gateway!
+
+First, we'll need to take care of a few bootstrapping items needed for our Go
+program. Here we initialize this program's Go modules file needed for tracking
+dependencies. Then, we fetch the needed deps.
+
+```console
+$ go mod init
+go: creating new go.mod: module github.com/brigadecore/brigade/examples/gateways/example-gateway
+go: to add module requirements and sums:
+	go mod tidy
+
+$ go mod tidy
```

Comment:
Can this command potentially be omitted?

iirc, most go commands, including `go run` will automatically perform dependency resolution.

Reply:
`go run` didn't do it for me (go 1.16.6) but I've replaced `go mod tidy` w/ the `go get github.com/brigadecore/brigade/sdk/v2` line since that's now the only dep for this program -- as it should be!